Source/Xamarin/Prism.Autofac.Forms/PrismApplication.cs

```diff
@@ -0,0 +1,125 @@
+        /// </summary>
+        const string _navigationServiceName = "AutofacPageNavigationService";
+
+        /// <summary>
+        /// Create a new instance of <see cref="PrismApplication"/>
+        /// </summary>
+        /// <param name="platformInitializer">Class to initialize platform instances</param>
+        /// <remarks>
+        /// The method <see cref="IPlatformInitializer.RegisterTypes(IContainer)"/> will be called after <see cref="PrismApplication.RegisterTypes()"/> 
+        /// to allow for registering platform specific instances.
+        /// </remarks>
+        public PrismApplication(IPlatformInitializer initializer = null) : base(initializer) { }
+
+
+        protected override void ConfigureViewModelLocator()
+        {
+            ViewModelLocationProvider.SetDefaultViewModelFactory((view, type) =>
+            {
+                NamedParameter parameter = null;
+                var page = view as Page;
+                if (page != null)
+                {
+                    var navService = CreateNavigationService();
+                    ((IPageAware)navService).Page = page;
+
+                    parameter = new NamedParameter("navigationService", navService);
+                }
+
+                return Container.Resolve(type, parameter);
+            });
+        }
+
+        /// <summary>
+        /// Create a default instance of <see cref="IContainer" />
+        /// </summary>
+        /// <returns>An instance of <see cref="IContainer" /></returns>
+        protected override IContainer CreateContainer()
+        {
+            return new ContainerBuilder().Build();
+        }
+
+        protected override IModuleManager CreateModuleManager()
+        {
+            return Container.Resolve<IModuleManager>();
+        }
+
+        /// <summary>
+        /// Create instance of <see cref="INavigationService"/>
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="_navigationServiceKey"/> is used as service key when resolving
+        /// </remarks>
+        /// <returns>Instance of <see cref="INavigationService"/></returns>
+        protected override INavigationService CreateNavigationService()
+        {
+            return Container.ResolveNamed<INavigationService>(_navigationServiceName);
+        }
+
+        /// <summary>
+        /// Create instance of <see cref="INavigationService"/> and set the <see cref="IPageAware.Page"/> property to <paramref name="page"/>
+        /// </summary>
+        /// <param name="page">Active page</param>
+        /// <returns>Instance of <see cref="INavigationService"/> with <see cref="IPageAware.Page"/> set</returns>
+        protected INavigationService CreateNavigationService(Page page)
+        {
+            var navigationService = CreateNavigationService();
+            ((IPageAware)navigationService).Page = page;
+            return navigationService;
+        }
+
+        protected override void InitializeModules()
```

Comment:
Just remove the test.  That method was added by the DryIoC package author just to test that scenario.